x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRankShardContext.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.rank.rrf;
+
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.elasticsearch.common.util.Maps;
+import org.elasticsearch.search.rank.RankShardContext;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Executes queries and generates results on the shard for RRF.
+ */
+public class RRFRankShardContext extends RankShardContext {
+
+    protected final int rankConstant;
+
+    public RRFRankShardContext(List<Query> queries, int from, int windowSize, int rankConstant) {
+        super(queries, from, windowSize);
+        this.rankConstant = rankConstant;
+    }
+
+    @Override
+    public RRFRankShardResult sort(List<TopDocs> rrfRankResults) {
```

Comment:
could we have unit tests for this sort method?

Reply:
Added.